src/index.js

```diff
@@ -92,13 +92,20 @@ class CID {
     return this.buffer
   }
 
+  toV0String () {
+    if (this.version === 0) {
+      return mh.toB58String(this.multihash)
+    }
+
+    throw new Error(`Not supported on version: ${this.version}`)
+  }
+
   /* defaults to base58btc */
   toBaseEncodedString (base) {
     base = base || 'base58btc'
 
     switch (this.version) {
       case 0:
-        return mh.toB58String(this.multihash)
       case 1:
```

Comment:
CIDV0 with `multibase` header is something that doesn't exist, if people need to represent CIDV0 with different bases, they should just update them to CIDV1, otherwise it is impossible to reason through code which base is CIDV0.